src/main/java/com/google/api/gax/grpc/ServiceApiSettings.java

```diff
@@ -200,14 +202,42 @@ public Builder provideChannelWith(final ManagedChannel channel, final boolean sh
     }
 
     /**
-     * Provides the connection settings necessary to create a channel.
+     * Provides the connection settings necessary to create a channel. This will overwrite any
+     * previous calls to provideChannelWith, including calls providing Credentials or a
+     * ManagedChannel.
      */
-    public Builder provideChannelWith(
-        final ConnectionSettings settings) {
+    public Builder provideChannelWith(final ConnectionSettings settings) {
       channelProvider = createChannelProvider(settings);
       return this;
     }
 
+    protected abstract ConnectionSettings getDefaultConnectionSettings();
```

Comment:
The implementation of getDefaultConnectionSettings in the XSettings.Builder should provide the static DEFAULT_CONNECTION_SETTINGS object on the XSettings class. This abstract method is there so that an instance on the builder can access the DEFAULT_CONNECTION_SETTINGS object. If we put the getDefaultConnectionSettings method on the XSettings class then the builder will need an instance of XSettings; if we make it static then we cannot override it.

Another alternative would be to implement the provideChannelWith methods in the settings.snip, where it can directly access the correct DEFAULT_CONNECTION_SETTINGS object.